homeassistant/components/automation/__init__.py

```diff
@@ -160,21 +160,21 @@ def setup(hass, config):
     @asyncio.coroutine
     def trigger_service_handler(service_call):
         """Handle automation triggers."""
-        for entity in component.extract_from_service(service_call):
+        for entity in component.async_extract_from_service(service_call):
             hass.loop.create_task(entity.async_trigger(
                 service_call.data.get(ATTR_VARIABLES), True))
 
     @asyncio.coroutine
     def turn_onoff_service_handler(service_call):
         """Handle automation turn on/off service calls."""
         method = 'async_{}'.format(service_call.service)
-        for entity in component.extract_from_service(service_call):
+        for entity in component.async_extract_from_service(service_call):
             hass.loop.create_task(getattr(entity, method)())
 
     @asyncio.coroutine
```

Comment:
@pvizeli pointed out to me that we use create_task so we can use callbacks. Fixed.